src/lib/proof-system/zkprogram.ts

```diff
@@ -1014,11 +1139,19 @@ function picklesRuleFromFunction(
       let computedTag: unknown;
       // Only create the tag if it hasn't already been created for this specific Proof class
       if (SideloadedTag.get(tag.name) === undefined) {
+        let featureFlags = [
+          0,
+          ...Object.entries(Proof.featureFlags).map(([_, flag]) =>
+            MlOption.mapTo(flag, MlBool)
+          ),
```

Comment:
we can't rely on the order of keys in `Proof.featureFlags` here, because it can be user input. I would make a second version of `featureFlagsToMl` below with a different return type